libraries/aws_nat_gateways.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsNatGateways < AwsResourceBase
+  name 'aws_nat_gateways'
+  desc 'Verifies settings for a collection of AWS NAT gateways.'
+
+  example "
+    describe aws_nat_gateways do
+      its('count') { should eq 3 }
+    end
+
+    describe aws_nat_gateways.where(name: 'my-gateway') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:ids, field: :id)
+             .register_column(:names, field: :name)
+             .register_column(:tags, field: :tags)
+             .register_column(:vpc_ids, field: :vpc_id)
+             .register_column(:subnet_ids, field: :subnet_id)
+             .register_column(:states, field: :state)
+             .register_column(:nat_gateways, field: :nat_gateway)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def to_s
+    "NAT Gateway #{@display_name}"
+  end
+
+  def fetch_data
+    ngw_rows = []
+    query_params = {}
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.compute_client.describe_nat_gateways(query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
+
+      @api_response.nat_gateways.each do |ngw|
+        # Create additional methods defined in the singular resource for consistency.
+        ngw.define_singleton_method(:to_s) { "Nat Gateway #{ngw.nat_gateway_id}" }
+        ngw.define_singleton_method(:nat_gateway_address_set) { ngw.nat_gateway_addresses&.first&.to_h }
+        ngw.define_singleton_method(:id) { ngw.nat_gateway_id }
```

Comment:
Do we need to include these methods? They seem kind of redundant if there's more than one network gateway in the response. Seems like the sort of thing the user would want to call from the singular resource only.

Reply:
By adding these methods, each gateway in the  `nat_gateways` column will be able to respond to the same tests applicable to the singular resource.
This will prevent having to make separate API calls by using the singular resource for the same data that has been acquired by the multiple resource.
I documented this option in the docs and integration tests, so I believe/hope users would adopt this approach.

Reply:
I'd prefer to avoid the approach of taking the `@api_response` sub-structure, creating singleton methods on it and passing the result into the FilterTable.  As an alternative, how about offering an option to instantiate the single resource with the result from the call in the constructor?  That way if anything changes in the methods for the single resource class you don't have to maintain the similar code in two places.  

As a second alternative, simply unpack the response and offer only a subset of those attributes for the plural resource - we don't strictly need to enforce any consistency between the plural/singular resource attributes and we can explore caching to avoid the additional callouts (outside of the scope of this PR).

Reply:
Removed `nat_gateway` object from the FilterTable because of the possible side effects on the used memory.